test/gtest/uct/test_md.cc

```diff
@@ -299,11 +299,15 @@ UCS_TEST_P(test_md, mem_type_detect_mds) {
 
         /* test mem_query API */
         uct_md_mem_attr_t mem_attr;
-        mem_attr.field_mask = UCT_MD_MEM_ATTR_FIELD_MEM_TYPE |
-                              UCT_MD_MEM_ATTR_FIELD_SYS_DEV;
+        mem_attr.field_mask = UCT_MD_MEM_ATTR_FIELD_MEM_TYPE     |
+                              UCT_MD_MEM_ATTR_FIELD_SYS_DEV      |
+                              UCT_MD_MEM_ATTR_FIELD_BASE_ADDRESS |
+                              UCT_MD_MEM_ATTR_FIELD_ALLOC_LENGTH;
         status = uct_md_mem_query(md(), address, buffer_size, &mem_attr);
         ASSERT_UCS_OK(status);
         EXPECT_EQ(alloc_mem_type, mem_attr.mem_type);
+        EXPECT_TRUE((mem_attr.alloc_length >= buffer_size) || (mem_attr.alloc_length == -1));
+        EXPECT_TRUE((mem_attr.base_address == NULL) || (mem_attr.base_address = address));
```

Comment:
1. == vs. =
2. pls really test it : allocate some buffer and check if pointer/size in middle of buffer can find the real address/length of the backing buffer

Reply:
1. Will fix!
2. Unfortunately I couldn't test as on the current machine, I have all gtests fail because max transports exceeded issue. And gtests don't accept UCX_TLS and I believe UCX_TLS is always hardcoded to `all`. The other machines where I have smaller number of NICs should be available soon. I'll try with change `1.` and test and push changes.

Reply:
Pushed changes to query at random offests and things work with CUDA. 
```
[----------] 1 test from cuda_cpy/test_md
[ RUN      ] cuda_cpy/test_md.mem_type_detect_mds/0 <cuda_cpy>
[     INFO ] cuda: sys_dev[] (0000:06:00.0)
[     INFO ] cuda-managed: sys_dev[] (0000:06:00.0)
[       OK ] cuda_cpy/test_md.mem_type_detect_mds/0 (147 ms)
[----------] 1 test from cuda_cpy/test_md (147 ms total)
```
ROCM md is untested though.

Reply:
2. can try with UCX_NET_DEVICES